crates/fluvio-storage/src/replica.rs

```diff
@@ -103,40 +103,17 @@ impl ReplicaStorage for FileReplica {
     /// return the size in byte
     #[instrument(skip(self))]
     async fn get_partition_size(&self) -> Result<u64, ErrorCode> {
-        let mut entries = read_dir(&self.option.base_dir).await.map_err(|err| {
-            error!(
-                "failed to open the log base dir \"{}\": {}",
-                self.option.base_dir.to_string_lossy(),
-                err
-            );
-            ErrorCode::StorageError
-        })?;
+        let active_len = self.active_segment.get_msg_log().get_pos();
 
-        let mut total_size = None;
-        while let Ok(Some(entry)) = entries.try_next().await {
-            let os_file_name = entry.file_name();
-
-            if os_file_name.to_string_lossy().ends_with(".log") {
-                let metadata = entry.metadata().await.map_err(|err| {
-                    error!(
-                        "fetching metadata for file \"{}\" failed: {err}",
-                        os_file_name.to_string_lossy(),
-                    );
-                    ErrorCode::StorageError
-                })?;
-                let file_len = metadata.len();
-                debug!("Log segment found. Add {file_len} byte to the total_size.");
-                *total_size.get_or_insert(0) += file_len;
-            }
-        }
+        debug!("Active segment length: {active_len}.");
+        let total_prev_segments_len = {
+            let reader = self.prev_segments.read().await;
+            reader.get_total_logs_len() as u32
```

Comment:
I didn't use atomic here, as advised, since the bump of `fluvio-future` did solve my issue.
Do you want me to change the implementation?

Reply:
No this is fine.